akka-persistence-typed/src/test/scala/akka/persistence/typed/scaladsl/DeletionApiTryout.scala

```diff
@@ -0,0 +1,58 @@
+/**
+ * Copyright (C) 2017-2018 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.persistence.typed.scaladsl
+
+import akka.Done
+import akka.actor.typed.{ActorRef, Behavior}
+import akka.persistence.SnapshotSelectionCriteria
+import akka.persistence.typed.scaladsl.PersistentBehaviors.CommandHandler
+
+import scala.concurrent.Future
+
+object DeletionApiTryout {
+
+  object State {
+    val empty = State(0)
+  }
+
+  final case class State(state: Int) {}
+
+  sealed trait Command
+  final case class Modify(n: Int) extends Command
+
+  case class Modified(mod: Int)
+
+  private def commandHandler: CommandHandler[Command, Modified, State] =
+    (ctx, state, cmd) ⇒
+      cmd match {
+        case Modify(n) ⇒
+          Effect.persist(Modified(n))
+      }
+
+  private def eventHandler(state: State, event: Modified): State =
+    state.copy(state = state.state + event.mod)
+
+  def behavior(entityId: String): Behavior[Command] =
+    PersistentBehaviors.receive[Command, Modified, State](
+      persistenceId = "State-" + entityId,
+      initialState = State.empty,
+      commandHandler,
+      eventHandler
+    ).snapshotEvery(5)
+     .afterSnapshot { (ctx, state, event, persistenceId, sequenceNumber) =>
+       // purely side-effect stuff here - feels bad, but alternative would be a declarative
+       // way of describing what deletions to perform - which also feels bad
```

Comment:
The reason why it is ugly is that it all of a sudden introduced imperative logic in an otherwise declarative/FP:y API.

What about a separate little DSL of snapshotting choices perhaps?
instead of `.snapshot(5)` and `.snapshotWhen(...)` something like
`.withSnapshots(Snapshot.every(5).keep(2).deleteEvents())`

Reply:
Perhaps still keeping the management versions for out-of-persistent-actor-cleanup work?